### PR TITLE
Rename primitive -> scalar and replace keywordp with scalarp.

### PR DESCRIPTION
**Pull request by bkuehnert**
This PR changes the convention of calling low-level types "primitive" to calling them "scalar", which is consistent with the public protobuf documentation.

It also replaces the "keywordp" function (which was used as a way of checking if a certain class identifier named a scalar type) with the function "scalarp", which checks if the type in question is a member of the list of scalar types.

### CODE REVIEW

File: parser.lisp [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 (in-package "PROTO-IMPL")
 
-(defparameter +PRIMITIVE-TYPES+
+(defparameter +SCALAR-TYPES+
```
Comment (Slids) [inline parser.lisp:9]: defconstant?

Comment (bkuehnert) [reply parser.lisp:9]: I suppose so

Comment (Slids) [reply parser.lisp]: Please remember to resolve.

**Comment by Slids** [comment]
Warning:
CheckLint found errors.
No mention of parameter TYPE in docstring for function SCALARP. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/clos-transform.lisp:199
No mention of parameter BUFFER in docstring for function SERIALIZE-SCALAR. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/wire-format.lisp:166
No docstring for function GET-SCALAR-ENCODER-FORM. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/wire-format.lisp:202
No docstring for function GET-SCALAR-ENCODER-LAMBDA. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/wire-format.lisp:220
No mention of parameter INDEX in docstring for function DESERIALIZE-SCALAR. (MISSING-DOCUMENTATION)
	//depot/google3/third_party/lisp/cl_protobufs/wire-format.lisp:549

File: api.lisp [hunk 2]
```diff
@@ -40,11 +40,11 @@ The definition of initialized is all required-fields are set."
...
 
-(defun primitive-field-equal (object-1 object-2)
-  "Check if two primitive objects in proto-fields are equal.
+(defun scalar-field-equal (object-1 object-2)
+  "Check if two objects with scalar type in proto-fields are equal.
 Parameters:
-  OBJECT-1: The first primitive object.
-  OBJECT-2: The second primitive object."
+  OBJECT-1: The first scalar object.
+  OBJECT-2: The second sclaar object."
```
Comment (cgay) [inline api.lisp:47]: typo

File: api.lisp [hunk 3]
```diff
@@ -40,11 +40,11 @@ The definition of initialized is all required-fields are set."
             "There is no Protobufs message for the class ~S" class)
     (object-initialized-p object message)))
 
-(defun primitive-field-equal (object-1 object-2)
-  "Check if two primitive objects in proto-fields are equal.
+(defun scalar-field-equal (object-1 object-2)
+  "Check if two objects with scalar type in proto-fields are equal.
```
Comment (cgay) [inline api.lisp:44]: pre-existing: what does "in proto-fields" mean in this sentence?

File: clos-transform.lisp [hunk 4]
```diff
@@ -196,6 +196,17 @@
 (defun lisp-type-to-protobuf-class (type)
   "Return protobuf class associated with the lisp type TYPE."
   (nth-value 1 (lisp-type-to-protobuf-type type)))
+(defun scalarp (type)
+
+
```
Comment (cgay) [inline clos-transform.lisp:201]: Remove extra blank lines.  Add blank line above the defun.

File: clos-transform.lisp [hunk 5]
```diff
@@ -196,6 +196,17 @@
...
+  "Returns true if the given Protobufs type is a scalar type. Scalar types
+are defined by the protobuf documentation. The symbol type is included here,
+as it is just a renaming of the string type.
```
Comment (cgay) [inline clos-transform.lisp:204]: You need to mention TYPE in uppercase or you'll get a lint error internally.

I don't follow the bit about symbol being included. Do you mean that `:symbol` is a synonym for `:string` and if so, why?

File: text-format.lisp [hunk 6]
```diff
@@ -315,13 +318,15 @@ Parameters:
...
+                            (cond ((scalarp type)
                                    (expect-char stream #\:)
                                    (let ((val (case type
                                                 ((:float) (parse-float stream))
                                                 ((:double) (parse-double stream))
                                                 ((:string) (parse-string stream))
-                                                ((:bool)   (if (boolean-true-p (parse-token stream)) t nil))
+                                                ((:bool)   (if (boolean-true-p (parse-token stream))
+                                                               t
+                                                               nil))
```
Comment (cgay) [inline text-format.lisp:329]: All of the `(if (boolean-true-p ...) t nil)` can be reduced to `(boolean-true-p ...)`

**Comment by Slids** [comment]
Good note by Carl, the linter requires a docstring to have all of the paramters mentioned  
and in upper case (at least once).

File: api.lisp (hunk 3 above)
Comment (bkuehnert) [reply api.lisp:44]: I don't think it means anything at all. The function is not proto specific, so I don't see why we should mention it.

**Pull request merged**